app/models/manageiq/providers/amazon/agent_coordinator_worker/runner/response_thread.rb

```diff
@@ -0,0 +1,187 @@
+module ManageIQ::Providers::Amazon::AgentCoordinatorWorker::Runner::ResponseThread
+  require 'xml/xml_utils'
+  require 'scanning_mixin'
+  include ScanningMixin
+
+  #
+  # This thread startup method is called by the AgentCoordinatorWorker::Runner
+  # at the end of the do_before_work_loop method.
+  #
+  def start_response_thread
+    Thread.new do
+      _log.info("Starting AWS SSA Response Thread")
+      begin
+        # Wait a bit for the Agent Coordinators to be started
+        sleep(20)
+        response_handler_loop
+      rescue SystemExit
+        _log.info("SystemExit received, exiting AWS SSA Response Thread")
+        stop_response_thread
+        Thread.exit
+      rescue => err
+        _log.error("start_response_thread: Error [#{err}]")
+        _log.error("start_response_thread: Error - [#{err.backtrace.join("\n")}]")
+      end
+    end
+  end
+
+  def stop_response_thread
+    @shutdown_instance_wait_thread = true
+  end
+
+  def response_handler_loop
+    @shutdown_instance_wait_thread = nil
+    until @shutdown_instance_wait_thread
+      @coordinators.each do |coord|
```

Comment:
Is there a possibility of cross-thread races with the coordinator modifying the shared `@coordinators` and you using them?  I wonder if there should be a mutex around access to `@coordinators`...probably better to use a attr_accessor method at a minimum than hitting the instance variable directly.

Reply:
Fixed in https://github.com/ManageIQ/manageiq-providers-amazon/pull/360 via a mutex.